### ISSUE TITLE
Fix our export filenames - Export source should not be added with spaces to the file name

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2929]
Currently, outputs like `Multicut Segmentation` and `Simple Segmentation` will be added to the file names (per default) via the `{export_source}` placeholder. In the aforementioned cases, this adds spaces to the file names, which is a big bummer, and not good practice. If someone were to ask me how to not name their files, this would be one of the points I'd say.

We could fix that by adding e.g. hyphens in case of those multi-word export sources. Downside: people might rely on the names for some down stream processing.

**Comment by k-dominik** [comment]
related: https://github.com/ilastik/ilastik/issues/2653 (options shown in error case should be the same as in documentation)

**Comment by btbest** [comment]
I agree with the problem you mention - I'd be too worried that to circumvent the issue with spaces in the file names, people have hard-coded the specific file names that ilastik will produce. To me, that would make hyphenation-by-default ask for a minor version bump instead of patch only.

We could alternatively introduce a new magic placeholder like `{export_source_hyphenated}` and put that new placeholder in the default export path format. That way existing projects should keep the old placeholder and maintain the same output file names.

**Comment by SurbhiD404** [comment]
Hi, I'd like to work on this issue.

I've investigated the export filename generation path and traced it through the codebase:

* `multicutWorkflow.py` and `edgeTrainingWithMulticutWorkflow.py` define `EXPORT_NAMES = ["Multicut Segmentation"]`
* `thresholdMaskingWorkflow.py` defines export names such as `"Thresholded Output"` and `"Inverted Thresholded Output"`
* These values are passed to `SelectionNames`
* `SelectionNames` becomes `result_type` in `DataExportPathFormatter`
* The default export filename template in `config.py` is `{dataset_dir}/{nickname}_{result_type}`

This means workflow export names containing spaces propagate directly into the default export filename template. Since several workflows define export names with spaces, the resulting default filenames can also contain spaces.

I plan to implement the backward-compatible approach suggested by @btbest: add a new `{export_source_hyphenated}` placeholder that replaces spaces with hyphens while preserving the existing `{result_type}` behavior, and update the default template to use the new placeholder.

Would it be okay if I work on this?

**Comment by SurbhiD404** [comment]
I've opened a PR implementing the `export_source_hyphenated` placeholder approach discussed above:
PR #3229
The change preserves `{result_type}` for backward compatibility, updates the default template to use `{export_source_hyphenated}`, and includes formatter tests covering the new placeholder.
